src/interpret.c

```diff
@@ -4221,6 +4221,15 @@ class Interpreter : public Visitor
 
     void visit(BinAssignExp *e)
     {
+        if (goal == ctfeNeedLvalue)
+        {
+            Expression *e1 = e->e1;
+            while (e->e1->op == TOKcast)
+                e1 = ((CastExp *)e1)->e1;
+            result = interpret(e1, istate, goal);
```

Comment:
You're right. I filed the new CTFE issue: https://issues.dlang.org/show_bug.cgi?id=14371